retis-events/src/compat/compat.rs

```diff
@@ -0,0 +1,138 @@
+#![allow(dead_code)] // Not everything is used in all versions.
+
+use anyhow::Result;
+
+/// When making breaking changes in the event, fixups should be added.
+use CompatFixup::*;
+const FIXUPS: &[&[CompatFixup]] = &[
+    /* CompatVersion::V0 */
+    &[],
+    /* CompatVersion::V1 */
+    &[
+        Add("ct/ct_status", CompatValue::Uint(0)),
+        // Strictly speaking we would need the following, however this
+        // intermediate state was never released so we can skip it.
+        // `Move("skb/packet/packet", "skb/packet/raw")`
+        Move("skb/ns/netns", "skb/ns/inum"),
+        Move("skb/packet", "packet"),
+        Move("skb/dev", "dev"),
+        Move("skb/ns", "netns"),
+        Move("packet/raw", "packet/data"),
+    ],
+];
+
+enum CompatFixup<'a> {
+    Remove(&'a str),
+    Add(&'a str, CompatValue),
+    Move(&'a str, &'a str),
+}
+
+pub(crate) enum CompatStrategy {
+    Backward(CompatVersion /* from */),
+    Forward(CompatVersion /* to */),
+}
+
+/// Representation of the event version we should be compatible with. This is an
+/// internal only representation and is usually derived from the Retis version
+/// itself. Multiple Retis versions can be translated to the same internal
+/// compatibility version, if no breaking change was made.
+///
+/// When breaking changes are made to the event in a given y-stream version,
+/// this should be reflected here.
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub(crate) enum CompatVersion {
+    /* v1.5.x; we start the backward compatibility effort with v1.5.0 */
+    V0 = 0,
+    /* v1.6.x.. */
+    V1,
+}
+
+impl CompatVersion {
+    pub const UNKNOWN: Self = Self::V0;
+    pub const LATEST: Self = Self::V1;
+
+    /// Create a compatibility version representation given a Retis version.
+    pub(crate) fn from_retis_version(version: &str) -> Self {
+        let compat_version = match version {
+            x if x.starts_with("v1.6.") || (x.starts_with("v1.5.") && x.contains("-")) => {
```

Comment:
Is the goal to match on unreleased versions between 1.5 and 1.6?

Reply:
> Is the goal to match on unreleased versions between 1.5 and 1.6?

Yes.

> If so, how do we really know what `CompatVersion` they represent?

There will always be a gap (between the latest release and the commit introducing changes) but there's no easy way to test for such thing. As the development version in unreleased this should be fine. It's needed otherwise using the development branch won't work.